api/app/src/main/kotlin/packit/service/RolePermissionService.kt

```diff
@@ -81,4 +101,90 @@ class BaseRolePermissionService(
 
         return rolePermissionsToAdd
     }
+
+    override fun updatePacketReadPermissionOnRoles(
+        rolesToAdd: List<Role>,
+        rolesToRemove: List<Role>,
+        packetGroupName: String,
+        packetId: String?,
+    )
+    {
+        val permission = permissionService.getByName("packet.read")
+        val packet = packetId?.let { packetService.getPacket(it) }
+        val packetGroup = if (packetId == null)
+        {
+            packetGroupService.getPacketGroupByName(packetGroupName)
+        } else
+        {
+            null
+        }
+
+        removePermissionFromRoles(
+            rolesToRemove,
+            permission,
+            packet,
+            packetGroup
+        )
+        addPermissionToRoles(
+            rolesToAdd,
+            permission,
+            packet,
+            packetGroup
+        )
+        roleRepository.saveAll(rolesToAdd)
+    }
```

Comment:
How atomic is this whole process? If an error is thrown by `addPermissionToRoles`, would the removal still take place? Can we use a transaction to make sure this is an all-or-nothing? (I have no experience with JPA so I don't know what's atomic and what isn't)

Reply:
ahh yup good catch !!! actually it wasnt atomic I've added the @transactional so that should rollback if errors occcur